core/CronArchive.php

```diff
@@ -857,10 +857,11 @@ public function invalidateRecentDate($dateStr, $idSite)
             'date' => $date->getDatetime(),
         ]);
 
-        $this->invalidateWithSegments([$idSite], $date->toString(), 'day');
+        $isYesterday = $dateStr == 'yesterday';
+        $this->invalidateWithSegments([$idSite], $date->toString(), 'day', false, $doNotIncludeTtlInExistingArchiveCheck = $isYesterday);
```

Comment:
Note: this and below code skips a particular check if we are invalidating `yesterday` (not a date including `yesterday`, just `yesterday`; we do this like we're invalidating `today` in case the day has changed and we want to initiate archiving for yesterday).

Since we only care if the day changed since the archive was created, we want to check if there's any archive available that we can use, so we don't want to do a ttl check.

Reply:
could we maybe also add inline a comment for this explaining this behaviour?

Reply:
:+1: